src/main/java/com/mongodb/spark/sql/connector/config/AbstractMongoConfig.java

```diff
@@ -285,34 +284,27 @@ private MongoClientFactory getMongoClientFactory() {
         .filter(k -> k.startsWith(PREFIX))
         .forEach(
             k -> {
-              overridePrefix.ifPresent(
-                  prefix -> {
-                    if (k.startsWith(prefix)) {
-                      overrides.add(k);
-                    }
-                  });
-
-              if (ignorePrefixes.stream().noneMatch(k::startsWith)) {
+              if (k.startsWith(overridePrefix)) {
+                overrides.add(k);
+              } else if (!k.startsWith(ignorePrefix)) {
                 defaults.add(k);
               }
             });
 
     Map<String, String> usageSpecificOptions = new HashMap<>();
     // Add any globally scoped options
     addConnectionStringDatabaseAndCollection(PREFIX, caseInsensitiveOptions, usageSpecificOptions);
+
     defaults.forEach(
         k -> usageSpecificOptions.put(k.substring(PREFIX.length()), caseInsensitiveOptions.get(k)));
 
     // Add usage specifically scoped options
-    overridePrefix.ifPresent(
-        prefix -> {
-          addConnectionStringDatabaseAndCollection(
-              prefix, caseInsensitiveOptions, usageSpecificOptions);
-          overrides.forEach(
-              k ->
-                  usageSpecificOptions.put(
-                      k.substring(prefix.length()), caseInsensitiveOptions.get(k)));
-        });
+    addConnectionStringDatabaseAndCollection(
```

Comment:
Another improvement - the database and collection can be set just via the connection string.